sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ConditionalSimplification.cs

```diff
@@ -158,17 +158,74 @@ int Test(object a, object b, object y, bool condition)
             {
                 elem = (null);
             }
+        }
 
+        // we ignore lambdas because of type resolution for conditional expressions, see CS0173
+        Action LambdasAreIgnored(bool condition, object a, Action action)
+        {
             Action myAction;
-            if (false) // Noncompliant
+            if (false)
             {
                 myAction = () => { };
             }
             else
             {
                 myAction = () => { Console.WriteLine(); };
             }
+
+            if (condition)
+            {
+                return () => X();
+            }
+            else
+            {
+                return () => Y();
+            }
+
+            if (condition)
+            {
+                Task.Run(() => X());
+            }
+            else
+            {
+                Task.Run(() => Y());
+            }
+
+            if (condition)
+            {
+                Bar(s => true);
+            }
+            else
+            {
+                Bar(s => false);
+            }
+
+            // if one arg is lambda, ignore
+            if (condition)
+            {
+                Foo(1, "2", () => X());
+            }
+            else
+            {
+                Foo(1, "2", () => Y());
+            }
+
+            Action x;
+            if (action != null)
```

Comment:
this is for the check that suggests to use `??` , the above are for the ternary operator `?:` suggestions